File: apps/backend/services/Business/Business.Presentation/Endpoints/Assets/CreateAsset.Request.cs
namespace Business.Presentation.Endpoints.Assets;

/// <summary>JSON body for POST /assets.</summary>
public record CreateAssetRequest(
    string Name,
    Guid? CategoryId,
    string? Description,
    decimal? SalePrice,
    decimal? RentalPrice,
    int StockCount,
    string? ImgUrl);

File: apps/mobile/lib/core/features/inventory/models/requests/create_asset_request.dart
/// JSON body for POST /assets — mirrors the Business API's CreateAssetRequest.
class CreateAssetRequest {
  final String name;
  final String? categoryId;
  final String? description;
  final double? salePrice;
  final double? rentalPrice;
  final int stockCount;
  final String? imgUrl;

  const CreateAssetRequest({
    required this.name,
    required this.stockCount,
    this.categoryId,
    this.description,
    this.salePrice,
    this.rentalPrice,
    this.imgUrl,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'categoryId': categoryId,
        'description': description,
        'salePrice': salePrice,
        'rentalPrice': rentalPrice,
        'stockCount': stockCount,
        'imgUrl': imgUrl,
      };
}
